blueocean-dashboard/src/main/js/components/RunPipeline.jsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Simple widget for triggering the running of a pipeline.
+ */
+
+import React, { Component, PropTypes } from 'react';
+import Pipeline from '../api/Pipeline';
+import Branch from '../api/Branch';
+import { Toast } from '@jenkins-cd/design-language';
+
+export default class RunPipeline extends Component {
+
+    constructor(props) {
+        super(props);
+        const pipeline = new Pipeline(props.organization, props.pipeline);
+        this.branch = new Branch(pipeline, props.branch);
+        this.state = {
+            toast: undefined,
+        };
+    }
+
+    componentDidMount() {
+        const _this = this;
+        const reactContext = this.context;
+        const theBranch = this.branch;
+
+        this.branch.onJobChannelEvent((event) => {
+            if (event.jenkins_event === 'job_run_queue_enter') {
+                _this.setState({
+                    toast: { text: `Queued "${theBranch.name}"` },
+                });
+            } else if (event.jenkins_event === 'job_run_started') {
+                _this.setState({
+                    toast: {
+                        text: `Started "${theBranch.name}" #${event.jenkins_object_id}`,
+                        action: {
+                            label: 'Open',
+                            callback: () => {
+                                const runDetailsUrl = theBranch.runDetailsRouteUrl(event.jenkins_object_id);
+                                reactContext.location.pathname = runDetailsUrl;
+                                reactContext.router.push(runDetailsUrl);
+                            },
+                        },
```

Comment:
Yeah I agree but didn't see an obvious alternative.

In reality it seems to me like the `Toaster` thing you talked about is the right way to go. Having to add render components to popup a message/alert seems wrong to me.